### PR TITLE
RavenDB-18674 - Fix duplicate declared function name printing to log.

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18674/Wrong-query-expression-in-the-log

### Additional description

Fix duplicate 'declared function function_name function function_name' printing to admin log when query with declared function failed.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs [hunk 1]
```diff
@@ -110,10 +110,15 @@ public override void VisitOrderBy(List<(QueryExpression Expression, OrderByField
...
 
-        public override void VisitDeclaredFunction(StringSegment name, string func, DeclaredFunction.FunctionType type)
+        public override void VisitDeclaredFunction(string func, DeclaredFunction.FunctionType type)
         {
             EnsureLine();
-            _sb.Append("DECLARE function ").Append(name.Value).AppendLine(func).AppendLine();
+            _sb.Append("DECLARE");
+            if (func.StartsWith(' ') == false)
+            {
+                _sb.Append(" ");
```
Comment (aviv86) [inline src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs]:
```suggestion
                _sb.Append(' ');
```

File: test/SlowTests/Issues/RavenDB-18674.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                }
+            }
+
+            WaitForUserToContinueTheTest(store);
+        }
+
+        private class Item
+        {
+            public string Id { get; set; }
+            public string Content { get; set; }
```
Comment (aviv86) [inline test/SlowTests/Issues/RavenDB-18674.cs]: can remove everything after the `Assert.False`

File: src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs [hunk 3]
```diff
@@ -110,10 +110,15 @@ public override void VisitOrderBy(List<(QueryExpression Expression, OrderByField
             }
         }
 
-        public override void VisitDeclaredFunction(StringSegment name, string func, DeclaredFunction.FunctionType type)
+        public override void VisitDeclaredFunction(string func, DeclaredFunction.FunctionType type)
```
Comment (aviv86) [inline src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs]: why do we need the `type` here ? it seems like no one is using it

File: src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs (hunk 1 above)
Comment (shaharhikri) [reply src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs]: Done

File: test/SlowTests/Issues/RavenDB-18674.cs (hunk 2 above)
Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-18674.cs]: Done

File: src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs (hunk 3 above)
Comment (shaharhikri) [reply src/Raven.Server/Documents/Queries/AST/StringQueryVisitor.cs]: Done

**Review by aviv86 — APPROVED** [review]

**Pull request merged**